ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reflection/ReflectionIT.scala

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.sandbox.services.reflection
+
+import com.digitalasset.ledger.api.testing.utils.SuiteResourceManagementAroundAll
+import com.digitalasset.platform.sandbox.services.SandboxFixture
+import com.digitalasset.platform.testing.SingleItemObserver
+import io.grpc.reflection.v1alpha.{
+  ServerReflectionGrpc,
+  ServerReflectionRequest,
+  ServerReflectionResponse
+}
+import org.scalatest.{AsyncWordSpec, Matchers}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+
+final class ReflectionIT
+    extends AsyncWordSpec
+    with Matchers
+    with SandboxFixture
+    with SuiteResourceManagementAroundAll {
+
+  private val listServices: ServerReflectionRequest =
+    ServerReflectionRequest.newBuilder().setHost("127.0.0.1").setListServices("").build()
+
+  "Reflection service" when {
+
+    "accessed" should {
+
+      "provide a list of exposed services" in {
+        for {
+          response <- execRequest(listServices)
+        } yield {
+          response.getListServicesResponse.getServiceCount shouldEqual 13
+        }
+      }
+
+      "provide details about each service" in {
+        for {
+          servicesResponse <- execRequest(listServices)
+          symbolResponses <- Future.sequence {
+            servicesResponse.getListServicesResponse.getServiceList.asScala
+              .map(resp => execRequest(fileBySymbolReq(resp.getName)))
+          }
+        } yield {
+          for {
+            r <- symbolResponses
+            p <- r.getFileDescriptorResponse.getFileDescriptorProtoList.asScala
+          } {
+            // We filter for this string due to an exotic bug in the bazel-grpc setup, see grpc-definitions/BUILD.bazel.
+            assert(
+              !p.toStringUtf8.contains("bazel-out"),
+              s"filedescriptor ${p.toStringUtf8} contains string 'bazel-out'. This means grpc reflection will not work.")
+          }
+          all(symbolResponses) should have('hasErrorResponse (false))
+        }
+      }
+    }
+  }
+
+  private def fileBySymbolReq(symbol: String) =
+    ServerReflectionRequest.newBuilder().setFileContainingSymbol(symbol).build()
+
+  private def execRequest(request: ServerReflectionRequest) = {
+    val observer = new SingleItemObserver[ServerReflectionResponse](_ => true)
+    ServerReflectionGrpc.newStub(channel).serverReflectionInfo(observer).onNext(request)
+    observer.result.map(_.get)
```

Comment:
You can use `SingleItemObserver.first` here.

```scala
    SingleItemObserver[ServerReflectionResponse].first(
      ServerReflectionGrpc
        .newStub(channel)
        .serverReflectionInfo(_)
        .onNext(request)
    )

Reply:
Thanks, much better, I struggled with it for a few minutes and gave up. Thanks for keeping the quality up. :+1:

Reply:
Addressed by https://github.com/digital-asset/daml/pull/3582/commits/020dc833d392c091979e7e9552d935fd1a2fc125